code/go/0chain.net/smartcontract/storagesc/blobber.go

```diff
@@ -845,69 +849,111 @@ func (sc *StorageSmartContract) commitBlobberConnection(
 			"Invalid signature for write marker")
 	}
 
-	if blobAlloc.AllocationRoot == commitConnection.AllocationRoot && blobAlloc.LastWriteMarker != nil &&
-		blobAlloc.LastWriteMarker.PreviousAllocationRoot == commitConnection.PrevAllocationRoot {
-		return string(blobAllocBytes), nil
+	if len(commitConnection.ChainData)%32 != 0 {
+		return "", common.NewError("commit_connection_failed",
+			"Invalid chain data")
 	}
 
-	blobber, err := sc.getBlobber(blobAlloc.BlobberID, balances)
-	if err != nil {
-		return "", common.NewErrorf("commit_connection_failed",
-			"error fetching blobber: %v", err)
+	if len(commitConnection.ChainData) > (32 * MAX_CHAIN_LENGTH) {
+		return "", common.NewError("commit_connection_failed",
+			"Chain data length exceeds the maximum chainlength "+string(MAX_CHAIN_LENGTH))
 	}
 
-	if blobAlloc.Stats.UsedSize == 0 {
-		blobAlloc.LatestFinalizedChallCreatedAt = commitConnection.WriteMarker.Timestamp
-		blobAlloc.LatestSuccessfulChallCreatedAt = commitConnection.WriteMarker.Timestamp
+	if commitConnection.WriteMarker.ChainSize < 0 {
+		return "", common.NewError("commit_connection_failed",
+			"Invalid chain size")
 	}
 
-	changeSize := commitConnection.WriteMarker.Size
-	var prevWmSize int64
-
-	blobberAllocSizeBefore := blobAlloc.Stats.UsedSize
-	if isRollback(commitConnection, blobAlloc.LastWriteMarker) {
-		changeSize = -blobAlloc.LastWriteMarker.Size
+	var (
+		changeSize int64
+		prevWmSize int64
+	)
+	//branch logic according to ChainHash being present or not
+	if commitConnection.WriteMarker.ChainHash != "" {
+		if blobAlloc.AllocationRoot == commitConnection.AllocationRoot && blobAlloc.LastWriteMarker != nil &&
```

Comment:
please comment this logic, otherwise it will be difficult to figure out what's going on here next time

Reply:
Added comments for chain hash